conan/api/conan_api.py

```diff
@@ -69,8 +69,10 @@ def __init__(self, cache_folder=None):
         self.export = ExportAPI(self, self._api_helpers)
         self.remove = RemoveAPI(self)
         self.new = NewAPI(self)
-        self.upload = UploadAPI(self, self._api_helpers)
-        self.download = DownloadAPI(self)
+        #: Used to upload recipes and packages to remotes
+        self.upload: UploadAPI = UploadAPI(self, self._api_helpers)
```

Comment:
Is this typing really necessary? Can't the tools deduce the type automatically in this case? I mean, we have ``auto`` even in Java and C++ for long time, this seems a step backward in typing.